File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> dif = new HashMap<>();
        for(int i = 0; i < nums.length; i++) {
            dif.put(nums[i], i);
        }

        for(int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            var complementPosition = dif.get(complement);

            if (complementPosition != null && complementPosition != i) {
                return new int[] {i, complementPosition};
            }
        }
        return new int[0];
    }
}

File: Data Structures & Algorithms/is-anagram/submission-1.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        
        hashmap = {}

        for c in s:
            if c in hashmap:
                hashmap[c] += 1
            else:
                hashmap[c] = 1
        
        for c in t:
            if c not in hashmap:
                return False
            hashmap[c] -= 1
        
        for value in hashmap.values():
            if value != 0:
                return False
        
        return True
        


        
        